modules/tensor_mechanics/src/kernels/InertialForce.C

```diff
@@ -88,23 +102,87 @@ InertialForce::computeQpResidual()
     return _test[_i][_qp] * _density[_qp] *
            (accel + vel * _eta[_qp] * (1 + _alpha) - _alpha * _eta[_qp] * (*_vel_old)[_qp]);
   }
+
+  // Lumped mass option
+  // Only lumping the masses here
+  // will multiply by corresponding residual multiplier after lumping the matrix
+  else if (_time_integrator.isLumped() && _time_integrator.isExplicit())
+    return _test[_i][_qp] * _density[_qp];
+
+  // Consistent mass option
+  // Same for explicit, implicit, and implicit with HHT
   else
     return _test[_i][_qp] * _density[_qp] *
-           ((*_u_dotdot)[_qp] + (*_u_dot)[_qp] * _eta[_qp] * (1.0 + _alpha) -
+           ((*_u_dotdot_residual)[_qp] + (*_u_dot_residual)[_qp] * _eta[_qp] * (1.0 + _alpha) -
             _alpha * _eta[_qp] * (*_u_dot_old)[_qp]);
 }
 
+void
+InertialForce::computeResidual()
```

Comment:
@dschwen This is the type of madness you're trying to avoid per our conversation yesterday - correct? 

I promised @cbolisetti  I would clean this up some and forgot about it. Now it seems it's back up for discussion after the discussion yesterday.